### PR TITLE
Add a metric to capture how many times Dora falls back to UFS

### PR DESCRIPTION
**Pull request by ChunxuTang**
### What changes are proposed in this pull request?

 Add a metric to capture how many times Dora falls back to UFS when opening files.

### Why are the changes needed?

N/A

### Does this PR introduce any user facing changes?

N/A

### CODE REVIEW

**Comment by ChunxuTang** [comment]
alluxio-bot, check this please

**Comment by ChunxuTang** [comment]
alluxio-bot, check this please

File: core/client/fs/src/main/java/alluxio/client/file/DoraCacheFileSystem.java [hunk 1]
```diff
@@ -110,7 +115,9 @@ public FileInStream openFile(URIStatus status, OpenFilePOptions options)
               .build();
       return mDoraClient.getInStream(status, openUfsBlockOptions);
     } catch (RuntimeException ex) {
-      LOG.debug("Dora client open file error. Fall back to UFS.", ex);
+      UFS_FALLBACK_COUNTER.inc();
```
Comment (huanghua78) [inline core/client/fs/src/main/java/alluxio/client/file/DoraCacheFileSystem.java:118]: How about the GetStatus()?

Comment (ChunxuTang) [reply core/client/fs/src/main/java/alluxio/client/file/DoraCacheFileSystem.java:120]: Hi @huanghua78, thanks for your suggestion!

Are you referring to the UFS fallback in the `getStatus()` function? I updated the PR and added the capture of the UFS fallback there with the same metric (We can have another metric for the `getStatus()` if necessary). Feel free to have a look and let me know your thought.

File: core/common/src/main/java/alluxio/metrics/MetricKey.java [hunk 2]
```diff
@@ -2426,6 +2426,13 @@ public static String getSyncMetricName(long mountId) {
           .setIsClusterAggregated(false)
           .build();
 
+  public static final MetricKey CLIENT_UFS_FALLBACK_COUNT =
+      new Builder("Client.UfsFallBackCount")
+          .setDescription("The number of fallbacks to UFS in Dora when opening files.")
```
Comment (LuQQiu) [inline core/common/src/main/java/alluxio/metrics/MetricKey.java:2431]: Dora is an internal name not external name.
The number of fallbacks to UFS when failing to open file in Alluxio distributed cache

**Review by LuQQiu — COMMENTED** [review]
Leave minor comments, otherwise LGTM

File: core/client/fs/src/main/java/alluxio/client/file/DoraCacheFileSystem.java (hunk 1 above)
Comment (huanghua78) [reply core/client/fs/src/main/java/alluxio/client/file/DoraCacheFileSystem.java:120]: Yes, that's what I meant.
Looks good.

**Review by huanghua78 — APPROVED** [review]
Looks good.

File: core/common/src/main/java/alluxio/metrics/MetricKey.java (hunk 2 above)
Comment (ChunxuTang) [reply core/common/src/main/java/alluxio/metrics/MetricKey.java]: Gotcha! I updated the description based on your comment. Feel free to let me know whether it looks good to you.

**Review by LuQQiu — APPROVED** [review]
LGTM

**Comment by LuQQiu** [comment]
alluxio-bot, merge this please

**Pull request merged**